lib/sshutils/x11/display.go

```diff
@@ -86,13 +88,17 @@ func (d *Display) Dial() (XServerConn, error) {
 	return nil, trace.NewAggregate(unixErr, tcpErr)
 }
 
-// Listen opens an XServer listener
+// Listen opens an XServer listener. It will attempt to listen on the display
+// address for both tcp and unix and return an aggregate error, unless one
+// results in an addr in use error.
 func (d *Display) Listen() (XServerListener, error) {
 	unixSock, unixErr := d.unixSocket()
 	if unixErr == nil {
 		var l *net.UnixListener
 		if l, unixErr = net.ListenUnix("unix", unixSock); unixErr == nil {
 			return &xserverUnixListener{l}, nil
+		} else if errors.Is(unixErr, syscall.EADDRINUSE) {
```

Comment:
@Joerger I'm not sure I fully understand how this fixes the issue with concurrent sessions tbh. Previously, we would still return any error returned by Listen* call below as a part of the aggregate error. With this fix seems like we just return an error sooner. I'm probably missing something, could you explain?

Reply:
The function that calls `display.Listen` is `x11.OpenXServerListener`, which expects `EADDRINUSE` errors for x11 displays which are already being used. However, `display.Listen` was returning an aggregate error which does not resolve to `syscall.EADDRINUSE`. This change fixes that by returning the `syscall.EADDRINUSE` error immediately, not in aggregate.

[```
for displayNumber := displayOffset; displayNumber <= maxDisplay; displayNumber++ {
	display := Display{DisplayNumber: displayNumber}
	if l, err := display.Listen(); err == nil {
		return l, display, nil
	} else if !errors.Is(err, syscall.EADDRINUSE) {
		return nil, Display{}, trace.Wrap(err)
	}
}
```](https://github.com/gravitational/teleport/blob/ea810d30d99f26e58a190edc5facfbe0c09ea5e5/lib/sshutils/x11/conn.go#L81-L88)

Reply:
Okay, makes sense.